bin/objectbox_model_generator/lib/src/generator.dart

```diff
@@ -1,122 +1,128 @@
-        var props = [];
-        String idPropertyName;
-        for(var f in element.fields) {
-            if(f.metadata == null || f.metadata.length != 1)        // skip unannotated fields
-                continue;
-            var annotElmt = f.metadata[0].element as ConstructorElement;
-            var annotType = annotElmt.returnType.toString();
-            var annotVal = f.metadata[0].computeConstantValue();
-            var fieldTypeObj = annotVal.getField("type");
-            int fieldType = fieldTypeObj == null ? null : fieldTypeObj.toIntValue();
-
-            var prop = {
-                "name": f.name,
-                "id": annotVal.getField("id").toIntValue(),
-                "uid": annotVal.getField("uid").toIntValue(),
-                "flags": 0,
-            };
-
-            if(annotType == "Id") {
-                if(idPropertyName != null)
-                    throw InvalidGenerationSourceError("in target ${elementBare.name}: has more than one properties annotated with @Id");
-                if(fieldType != null)
-                     throw InvalidGenerationSourceError("in target ${elementBare.name}: programming error: @Id property may not specify a type");
-                if(f.type.toString() != "int")
-                    throw InvalidGenerationSourceError("in target ${elementBare.name}: field with @Id property has type '${f.type.toString()}', but it must be 'int'");
-
-                fieldType = OBXPropertyType.Long;                
-                prop["flags"] = OBXPropertyFlag.ID;
-                idPropertyName = f.name;
-            } else if(annotType == "Property") {
-                // nothing special here
-            } else {
-                // skip unknown annotations
-                continue;
-            }
-
-            if(fieldType == null) {
-                var fieldTypeStr = f.type.toString();
-                if(fieldTypeStr == "int")
-                    fieldType = OBXPropertyType.Int;
-                else if(fieldTypeStr == "String")
-                    fieldType = OBXPropertyType.String;
-                else {
-                    print("warning: skipping field '${f.name}' in entity '${element.name}', as it has the unsupported type '$fieldTypeStr'");
-                    continue;
-                }
-            }
-
-            prop["type"] = fieldType;
-            props.add(prop);
-            ret += """
-                {
-                    "name": "${prop['name']}",
-                    "id": ${prop['id']},
-                    "uid": ${prop['uid']},
-                    "type": ${prop['type']},
-                    "flags": ${prop['flags']},
-                    "flatbuffers_id": ${(prop['id'] as int) - 1},
-                },
-            """;
+import "code_chunks.dart";
+import "merge.dart";
+import "package:objectbox/src/modelinfo/index.dart";
+
+class EntityGenerator extends GeneratorForAnnotation<obx.Entity> {
+  static const ALL_MODELS_JSON = "objectbox-model.json";
```

Comment:
In Java this is `objectbox-models/default.json`. Is that name/path in line with Go?

Reply:
yeah, right now it's in line with Go, it may change in the future (swift does have a different name/path as well)